doc/manual/src/glossary.md

```diff
@@ -216,6 +216,18 @@
 
   [output path]: #gloss-output-path
 
+- [derived path]{#gloss-derived-path}
+
+  A [store]-level expression (not Nix language expression) that denotes a [store object].
+  There are two forms:
+
+  - *constant*: A constant derived path is just a [store path]
+
+  - *output*: An output derived path is a pair of a [store path] to a [derivation] and an [output] name.
+
+  Derived paths are necessary because in general, we do not know what the [output path] path of an [output] will buntil it is [realise]d.
+  We need a way to refer to unrelised outputs in order to be able to tell Nix to realise them!
```

Comment:
```suggestion
  Derived paths are necessary because in general, we do not know what the [output path] path of an [output] will be until it is [realise]d.
  Derived paths provide a way to refer to unrealised outputs in order to be able to tell Nix to realise them.
```